test/bin/pin-depends.t/run.t

```diff
@@ -0,0 +1,66 @@
+We have a simple project with a single package defined at the root.
+It has a `x-opam-monorepo-opam-repositories` field set to use a local
+opam-repository for locking and a pinned package.
+
+  $ cat pin-depends.opam
+  opam-version: "2.0"
+  depends: [
+    "dune"
+    "b"
+    "c"
+  ]
+  pin-depends: ["b.dev" "./duniverse/b"]
+  x-opam-monorepo-opam-provided: ["b"]
+  x-opam-monorepo-opam-repositories: [
+    "file://$OPAM_MONOREPO_CWD/minimal-repo"
+    "file://$OPAM_MONOREPO_CWD/repo"
+  ]
+
+We provided a minimal opam-repository but locking should be successful.
+
+  $ gen-minimal-repo
+  $ opam-monorepo lock
+  ==> Using 1 locally scanned package as the target.
+  ==> Found 10 opam dependencies for the target package.
+  ==> Querying opam database for their metadata and Dune compatibility.
+  ==> Calculating exact pins for each of them.
+  ==> Wrote lockfile with 1 entries to $TESTCASE_ROOT/pin-depends.opam.locked. You can now run opam monorepo pull to fetch their sources.
+
+The lockfile should contain the base packages, dune and our 2 dependencies
+`b` and `c` which should be pulled in the duniverse
+
+  $ cat pin-depends.opam.locked | sed 's|file://.*/pin-depends.t/|file://$LOCAL_PATH/|'
```

Comment:
Could you make this print just the relevant subsets of the lockfile? Otherwise the test is full of incidental content of the lockfile that will need to keep getting updated.

Something like `opam show --just-file --raw -fpin-depends ./pin-depends.opam.locked`.